descartes_light/include/descartes_light/interface/edge_evaluator.h

```diff
@@ -32,13 +32,59 @@ template <typename FloatType>
 class EdgeEvaluator
 {
 public:
+  using Ptr = typename std::shared_ptr<EdgeEvaluator<FloatType>>;
+  EdgeEvaluator(std::size_t dof) : dof_(dof) {}
   virtual ~EdgeEvaluator() {}
 
   virtual bool evaluate(const Rung_<FloatType>& from,
                         const Rung_<FloatType>& to,
-                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges) = 0;
+                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges)
+  {
+    const auto n_start = from.data.size() / dof_;
+    const auto n_end = to.data.size() / dof_;
 
-  typedef typename std::shared_ptr<EdgeEvaluator<FloatType>> Ptr;
+    // Allocate if needed because of Compound Evaluator)
+    if (edges.size() == 0)
+      edges.resize(n_start);
+
+    assert(edges.size() == n_start);
+
+    for (std::size_t i = 0; i < n_start; ++i)
+    {
+      const auto* start_vertex = from.data.data() + dof_ * i;
+      for (std::size_t j = 0; j < n_end; ++j)
+      {
+        const auto* end_vertex = to.data.data() + dof_ * j;
+
+        // Consider the edge:
+        auto results = considerEdge(from, start_vertex, to, end_vertex);
+        if (results.first)
+          edges[i].emplace_back(results.second, j);
+      }
+    }
+
+    for (const auto& rung : edges)
+      if (!rung.empty())
+        return true;
+
+    return false;
+  }
+
+  /**
+   * @brief This evalutes the edge and determines if it should be considered.
+   * @param from The rung associated with the start state
+   * @param start The start state of the edge
+   * @param to The rung associated with the end state
+   * @param end The end state of the edge
+   * @return A pair <True/False, Cost>
+   */
+  virtual std::pair<bool, FloatType> considerEdge(const Rung_<FloatType>& from,
```

Comment:
If this is a new function, you might consider renaming to something more descriptive like `computeEdgeValidity` or something like that. "Consider" seems pretty ambiguous to me

Also, do you really need access to the rungs? It seems like this function really only needs to know about the two vertices and the timing constraint between them.

Reply:
> Also, do you really need access to the rungs? It seems like this function really only needs to know about the two vertices and the timing constraint between them.

I originally did not have it but currently the timing constraint is tied to the rung. I would like to get ride of the timing constraint  in the rung because it should just be a separate edge evaluator. You good with removing it from the rung?

Reply:
I think separating the timing constraint from the the rung would be fine. It already comes into the `build` function as a [separate vector](https://github.com/swri-robotics/descartes_light/blob/e3f6d85063fea4ac77305ea4c180e7c151c1c374/descartes_light/include/descartes_light/descartes_light.h#L42) anyway